.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -0,0 +1,50 @@
+*Please fill the checklist so the group can get a quick understanding of your PR*
+### Category
+- [ ] **Feature**
+- [ ] **Bug Fix**
+- [ ] **Testing Oriented**
+- [ ] **Documentation**
+- [ ] **Other**
+### Domain
+- [ ] **Frontend**
+- [ ] **Backend**
+- [ ] **Both**
+
+### Description
+Include a summary of what this pull request is intended to achieve and key points you will end up addressing.
+
+### Related Issue(s)
+Make sure to reference an issues or pull requests that are related to your pull request in order to provide some context.
+
+### New Additions
+List any new file additions made to the project with a brief explanation of why your additons were necessary.
+This should match up with your "files changed" tab
+
+### Modifications
+List any files that were modified as part of your pull request and provide a quick explanation as to why the changes were necessary.
+This should match up with your "files changed" tab
+
+### New Dependencies
+Specifically list any new python packages that you needed to install or use, and any npm packages that were installed.
+This will give reviewers a better chance of spotting changes that need to be made to either `requirements.txt` or `package.json`
+
+### Visuals
+Include any screenshots or diagrams to help us better understand your additions/changes. visuals can be spread out throughout the PR.
+
+### Architecture of your changes 
+Here you can go more in depth and explain the underlying architecture of the system that you are introducing. If you are
+making changes to a previous system that just explain any differences. We should be able to understand your though process behind the
+code that you have written. 
+
+### Use of your changes
+Here you can explain how you expect the different systems of our app to interact with your new additions. You can also explain how us 
+as developers should be using your code. 
+
+### Testing
+Testing seems to be a very important part of this class. The prof wants us to be writing unit tests and I even think he expects us to
+apply TTD. We should at the very least describe what kind of testing we did to ensure that these new additions work properly. Also list out any
+edge cases you encountered or are suspicious of. Ideally we want to be writing unit tests and be able to provide the coverage here. If you
+have implemented unit test make sure to provide the commands necessary for others to run your tests.
+
+### Mentions
+Tag any group members that you wish to review your pr, helped with your pr, or need to take note of your pr.
```

Comment:
Capitalization of "pr" to make the sentence look like:

"Tag any group members that you wish to review your PR, helped with your PR, or need to take note of your PR."